From 6679affb13f08f8ff780a549da06cceb2694bbe7 Mon Sep 17 00:00:00 2001
From: Zeng Zhaoming <b32542@freescale.com>
Date: Thu, 21 Apr 2011 03:34:56 +0800
Subject: ENGR00142413: MX51: Fix mx51 I2c board info multiple defined in
 2.6.38

Multiple defined i2c board info in 2.6.38 kernel for mx51 bbg.
This cause a build error.

Signed-off-by: Zeng Zhaoming <b32542@freescale.com>
---
 arch/arm/mach-mx5/board-mx51_babbage.c |   13 ++++---------
 1 file changed, 4 insertions(+), 9 deletions(-)

diff --git a/arch/arm/mach-mx5/board-mx51_babbage.c b/arch/arm/mach-mx5/board-mx51_babbage.c
index 25a2fda..2d83b18 100755
--- a/arch/arm/mach-mx5/board-mx51_babbage.c
+++ b/arch/arm/mach-mx5/board-mx51_babbage.c
@@ -239,13 +239,6 @@ static const struct esdhc_platform_data mx51_bbg_sd2_data __initconst = {
 	.cd_gpio = MX51_BBG_SD2_CD,
 };
 
-static struct i2c_board_info mxc_i2c1_board_info[] __initdata = {
-	{
-	 .type = "sgtl5000",
-	 .addr = 0x0a,
-	 },
-};
-
 static void babbage_suspend_enter()
 {
 }
@@ -518,6 +511,10 @@ static struct fsl_mxc_ddc_platform_data bbg_ddc_dvi_data = {
 
 static struct i2c_board_info mxc_i2c1_board_info[] __initdata = {
 	{
+	 .type = "sgtl5000",
+	 .addr = 0x0a,
+	 },
+	{
 	 .type = "mxc_ddc",
 	 .addr = 0x50,
 	 .irq = gpio_to_irq(BABBAGE_DVI_DET),
@@ -677,8 +674,6 @@ static void __init mx51_babbage_init(void)
 	i2c_register_board_info(3, mxc_i2c_hs_board_info,
 				ARRAY_SIZE(mxc_i2c_hs_board_info));
 
-	i2c_register_board_info(1, mxc_i2c1_board_info,
-				ARRAY_SIZE(mxc_i2c1_board_info));
 	/*if (otg_mode_host)
 		mxc_register_device(&mxc_usbdr_host_device, &dr_utmi_config);
 	else {
-- 
1.7.9.5

